src/platform/site-wide/user-nav/containers/AutoSSO.jsx

```diff
@@ -0,0 +1,47 @@
+import React from 'react';
+import { connect } from 'react-redux';
+
+import { autoLogin, autoLogout } from 'platform/user/authentication/utilities';
+import { ssoe, ssoeInbound } from 'platform/user/authentication/selectors';
+import { hasSession, hasSessionSSO } from 'platform/user/profile/utilities';
+import { ssoKeepAliveSession } from 'platform/utilities/sso';
+
+class AutoSSO extends React.Component {
+  state = {
+    hasCalledKeepAlive: false,
+  };
+
+  componentDidUpdate() {
+    const { useSSOe, useInboundSSOe } = this.props;
+    const { hasCalledKeepAlive } = this.state;
+
+    if (useSSOe && useInboundSSOe && !hasCalledKeepAlive) {
+      this.checkStatus();
+    }
+  }
+
+  checkStatus = () => {
+    ssoKeepAliveSession().then(() => {
+      if (hasSession() && hasSessionSSO() === 'false') {
+        autoLogout();
+      } else if (!hasSession() && hasSessionSSO() === 'true') {
+        autoLogin();
+      }
+
+      this.setState({ hasCalledKeepAlive: true });
```

Comment:
Don't know how, but `hasCalledKeepAlive` seems like it could be stored or derived out of Redux, like added into state after the call to `autoLogin` or `autoLogout`

These conditions cover "true && false" and "false && true" but should there be an `else` or anything? If both are false or both true

Reply:
Those conditions are both inherently covered already:

* both true = already logged in across sessions, do nothing
* both false = no sessions active, do nothing.

Reply:
the redux point is interesting. would that be something added to `user.login` or something like that?

Reply:
Yeah, if these utilities can be dispatched as actions, I'd look at adding an `sso` reducer or something like that to this reducer - `src/platform/user/authentication/reducers/index.js`

Reply:
been a bit since I've touched the whole of a redux flow, but I made that refactor 😅